src/components/clusterConfiguration/ClusterPreflightRequirementsContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {
+  Cluster,
+  getClusterPreflightRequirements,
+  getErrorMessage,
+  handleApiError,
+  PreflightHardwareRequirements,
+} from '../../api';
+import { addAlert } from '../../features/alerts/alertsSlice';
+
+export type ClusterPreflightRequirementsContextType = {
+  preflightRequirements?: PreflightHardwareRequirements;
+};
+
+export const ClusterPreflightRequirementsContext = React.createContext<
+  ClusterPreflightRequirementsContextType
+>({});
+
+export const ClusterPreflightRequirementsContextProvider: React.FC<{ cluster: Cluster }> = ({
+  cluster,
```

Comment:
Looks like only `cluster.id` is used so maybe that could be enough to pass as a prop instead of the whole cluster.

Reply:
done